browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -102,20 +104,26 @@ void RewardsTabHelper::DidFinishLoad(
   rewards_service_->OnLoad(tab_id_, validated_url);
 }
 
-void RewardsTabHelper::DidFinishNavigation(content::NavigationHandle* handle) {
-  if (!handle->IsInMainFrame() || !handle->HasCommitted() ||
-      handle->IsDownload()) {
+void RewardsTabHelper::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle->HasCommitted() ||
+      !navigation_handle->IsInPrimaryMainFrame() || !rewards_service_) {
```

Comment:
no need to check IsDownload() anymore? is that covered by IsInPrimaryMainFrame() ?

Reply:
also there is no rewards_service_ for incognito tabs, is that correct?

Reply:
The documentation for `HasCommitted` suggests that it will not be true for downloads. However, just to be conservative I've added back the download check.

Yes - there's no RewardsService for incognito tabs.